File: testdata/bench/javascript/BenchmarkTest00002.js
// Safe: SQL injection prevented via parameterized query
const express = require('express');
const db = require('./db');
const router = express.Router();
router.get('/test002', (req, res) => {
    const param = req.query.input;
    db.query("SELECT * FROM users WHERE id = ?", [param], (err, results) => { res.json(results); });
});
module.exports = router;

File: testdata/bench/rust/BenchmarkTest00002.rs
use actix_web::{web, HttpResponse};
use std::collections::HashMap;

// Safe: SQL injection prevented with parameterized query bind
async fn handler(query: web::Query<HashMap<String, String>>) -> HttpResponse {
    let param = query.get("input").unwrap_or(&String::new()).clone();
    // sqlx::query("SELECT * FROM users WHERE id = $1").bind(&param).fetch_one(&pool).await
    HttpResponse::Ok().body("safe")
}
